### PR TITLE
[microNPU] Eliminate some compiler warnings for microNPU demo app

### PR DESCRIPTION
**Pull request by grant-arm**
* Eliminates all "control reaches end of non-void function" warnings
 * Eliminates all "unused variable" warnings
 * Eliminates some "implicit declaration of function" warnings

### CODE REVIEW

**Comment by grant-arm** [comment]
@Mousius @manupa-arm @areusch

**Comment by grant-arm** [comment]
@Mousius @manupa-arm @areusch this is passing again, so I would appreciated it if someone could take a look at it.

File: src/target/source/codegen_c_host.cc [hunk 1]
```diff
@@ -358,8 +370,18 @@ runtime::Module BuildCHost(IRModule mod, Target target) {
   using tvm::runtime::Registry;
   bool output_ssa = false;
   bool emit_asserts = false;
+
+  std::set<std::string> devices;
```
Comment (manupa-arm) [inline src/target/source/codegen_c_host.cc:374]: nit : unordered_set is preferred due to constant time .find()

File: src/target/source/codegen_c_host.cc [hunk 2]
```diff
@@ -43,14 +44,25 @@ namespace codegen {
...
+void CodeGenCHost::Init(bool output_ssa, bool emit_asserts, std::string target_str,
+                        const std::set<std::string>& devices) {
   emit_asserts_ = emit_asserts;
   declared_globals_.clear();
   decl_stream << "// tvm target: " << target_str << "\n";
   decl_stream << "#define TVM_EXPORTS\n";
   decl_stream << "#include \"tvm/runtime/c_runtime_api.h\"\n";
   decl_stream << "#include \"tvm/runtime/c_backend_api.h\"\n";
   decl_stream << "#include <math.h>\n";
+  if (devices.find("ethos-u") != devices.end()) {
```
Comment (manupa-arm) [inline src/target/source/codegen_c_host.cc:56]: nit : unordered_set is preferred due to constant time .find()

**Review by manupa-arm — COMMENTED** [review]
Broadly looks good.

I have left a nit use a better data structure, however, I admit that it does not have worrying scaling characteristic (i.e. fair to assume devices are going to be small). If you agree, we could do a follow up since this is green.

**Review by manupa-arm — APPROVED** [review]
Thanks @grant-arm !

**Pull request merged**